var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -83,16 +83,23 @@ class Hpx(CMakePackage, CudaPackage):
     # https://github.com/STEllAR-GROUP/hpx/issues/4728#issuecomment-640685308
     depends_on('boost@:1.72.0', when='@:1.4')
 
+    depends_on('boost@1.64.0:', when='@1.5')
+    depends_on('boost@1.66.0:', when='@1.6')
+
     depends_on('boost+filesystem', when='cxxstd=11')
     depends_on('boost+filesystem', when='cxxstd=14')
-    depends_on('boost+system+program_options+chrono+date_time+log+regex+thread')
+    depends_on('boost+filesystem', when='^gcc@:8')
+    depends_on('boost+filesystem', when='^clang@:8')
+    depends_on('boost+system', when='boost@:1.70')
```

Comment:
Is this valid? I couldn't actually get the `+system` variant in there when I concretized with boost 1.69 so I'm guessing it's not.

Reply:
I think it's testing the limits of the old concretizer :p it's self-referential? Let me test the clingo concretizer.

Reply:
Works with `clingo`:

```
hpx@1.6.0%gcc@10.2.0~async_cuda~async_mpi~cuda~examples~generic_coroutines~ipo~tools build_type=RelWithDebInfo cuda_arch=none cxxstd=17 instrumentation=none malloc=tcmalloc max_cpu_count=64 networking=tcp arch=linux-ubuntu20.04-zen2
    ^boost@1.69.0%gcc@10.2.0~atomic~chrono~clanglibcpp~container~context~coroutine~date_time~debug~exception~fiber~filesystem~graph~icu~iostreams~locale~log~math~mpi+multithreaded~numpy~pic~program_options~python~random~regex~serialization+shared~signals~singlethreaded+system~taggedlayout~test~thread~timer~versionedlayout~wave cxxstd=17 patches=3a83d907043708218325c35ffc318fd6d6cfd78ba89a78f2c70013c72603e5b8,607b0772dec1287c9084ae3b36ee32bff945a2fe5e608823ed47a1ea765c84cd visibility=hidden arch=linux-ubuntu20.04-zen2
```